### DISCUSSION TITLE
Question on Linking

### DISCUSSION THREAD

**Discussion by siddhsql** [k2-fsa/sherpa-onnx#1389] [Q&A]
Consider the `sherpa-onnx-jni` library. It has several dependencies like `piper_phonemize`, `espeak-ng` etc. 

/sherpa-onnx/build/_deps/piper_phonemize-build/CMakeFiles/piper_phonemize.dir/flags.make:

```
CXX_FLAGS =   -static-libstdc++ -static-libgcc  
```

/sherpa-onnx/build/_deps/espeak_ng-build/src/libespeak-ng/CMakeFiles/espeak-ng.dir/flags.make has no such entry.

I haven't investigated each and every dependency but I am trying to make sense of several questions:

1. How do you decide whether to include `-static-libstdc++ -static-libgcc` or not in a dependency?
2. Won't you end up with multiple copies of `libstdc++` and `libgcc` in final `sherpa-onnx-jni.so` file (one copy from each dep that had those flags in it, and a final additional copy from statically linking `sherpa-onnx-jni` itself against `libgcc` and `libstdc++`)  and won't it create all sorts of problems?

**Comment by csukuangfj** [comment]
What do you mean by "problems" here? Could you be more specific? If there are problems, are there any logs you can share with us?

If you think there are multiple copies of the libstdc++ in libsherpa-onnx-jni.so, could you show how you verify it? Are you able to get duplicate symbols from libsherpa-onnx-jni.so?
